Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java

```diff
@@ -0,0 +1,428 @@
+package edu.asu.conceptpower.servlet.core.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.servlet.core.ConceptEntry;
+import edu.asu.conceptpower.servlet.core.ConceptList;
+import edu.asu.conceptpower.servlet.core.IConceptManager;
+import edu.asu.conceptpower.servlet.db4o.DBNames;
+import edu.asu.conceptpower.servlet.db4o.IConceptDBManager;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryDoesNotExistException;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryEntryExistsException;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryModifyException;
+import edu.asu.conceptpower.servlet.exceptions.LuceneException;
+import edu.asu.conceptpower.servlet.lucene.ILuceneUtility;
+import edu.asu.conceptpower.servlet.rest.LuceneFieldNames;
+import edu.asu.conceptpower.servlet.rest.SearchFieldNames;
+import edu.asu.conceptpower.servlet.wordnet.Constants;
+import edu.asu.conceptpower.servlet.wordnet.WordNetManager;
+
+/**
+ * This class handles concepts in general. It uses a DB4O client that contains
+ * all added concepts and concept wrappers, and a WordNet client that queries
+ * WordNet.
+ * 
+ * @author Julia Damerow
+ * 
+ */
+@Service
+public class ConceptManager implements IConceptManager {
+
+    @Autowired
+    private IConceptDBManager client;
+
+    @Autowired
+    private ILuceneUtility luceneUtility;
```

Comment:
I'd use the search manager instead of connecting directly with the lucene utility and let the searchmanager manage the connection with lucene. This way, if we decide to switch to Solr or Elasticsearch or so later on, just one class has to be changed, not all of them. Or, thinking of it, you could also rename this class to Indexsoemthing (e.g. IndexUtility, or IndexService), so that this class could simply be replaced by a different implementation if we decide to change the indexer.